services/plugins/plugins/parsers.go

```diff
@@ -81,9 +83,24 @@ func (plugin *Plugin) persistFileSha(destPlugin *Plugin) error {
 	return nil
 }
 
-func (plugin *Plugin) PostCreateProcess(_ *provider.Client, destPluginRaw any) error {
+func (plugin *Plugin) PostCreateProcess(client *provider.Client, destPluginRaw any) error {
 	createdPlugin := destPluginRaw.(*Plugin)
-	plugin.persistFilePath(createdPlugin)
+
+	metadata, err := GetPluginMetadata(createdPlugin.ID, client)
+	if err != nil {
+		return plugin.persistFileSha(createdPlugin)
+	}
+
+	startTime := time.Now()
+
+	for metadata.Status != "ACTIVE" && time.Since(startTime) < 3*time.Minute {
```

Comment:
instead of 3 minutes we could use the timeout defined in [here](https://github.com/leanspace/terraform-provider-leanspace/blob/main/provider/client.go#L48), we could set it once and use it in this 2 files